src/main/main.cpp

```diff
@@ -7,16 +7,30 @@
 #include "util/Logging.h"
 
 #include "crypto/ByteSliceHasher.h"
+#include <cstdlib>
 #include <sodium/core.h>
 #include <xdrpp/marshal.h>
 
 INITIALIZE_EASYLOGGINGPP
 
+namespace stellar
+{
+static void
+outOfMemory()
+{
+    std::cerr << "Unable to allocate memory" << std::endl;
```

Comment:
This doesn't look as safe as I would like. Are you certain that memory allocation via `operator new` never occurs on this line? If not, this will trigger `outOfMemory` recursively and `std::abort()` will not be reached.

Reply:
we can just do fprintf/fflush over stderr as we don't do anything special